rust/datafusion/src/physical_plan/planner.rs

```diff
@@ -535,6 +536,32 @@ impl DefaultPhysicalPlanner {
                     input_schema,
                 )
             }
+            Expr::Between {
+                expr,
+                negated,
+                low,
+                high,
+            } => {
+                let value_expr =
+                    self.create_physical_expr(expr, input_schema, ctx_state)?;
+                let low_expr = self.create_physical_expr(low, input_schema, ctx_state)?;
+                let high_expr =
+                    self.create_physical_expr(high, input_schema, ctx_state)?;
+
+                // rewrite the between into the two binary operators
+                let binary_expr = binary(
+                    binary(value_expr.clone(), Operator::GtEq, low_expr, input_schema)?,
+                    Operator::And,
+                    binary(value_expr.clone(), Operator::LtEq, high_expr, input_schema)?,
+                    input_schema,
+                );
+
+                if *negated {
+                    expressions::not(binary_expr?, input_schema)
+                } else {
+                    binary_expr
+                }
+            }
```

Comment:
I like that we do not have to introduce a new physical op for this.

Reply:
I did not read the README correctly and so I will add a test in `tests/sql.rs` (I couldn't see the logical place for tests until you mentioned it).

Reply:
@jorgecarleitao I have added a test in `tests/sql.rs`.